example/src/screens/HybridObjectTestsScreen.tsx

```diff
@@ -179,8 +272,66 @@ export function HybridObjectTestsScreen() {
         <Text style={styles.buildTypeText}>{NitroModules.buildType}</Text>
       </View>
 
+      <View style={styles.searchContainer}>
+        <TextInput
+          style={[styles.searchInput, { color: colors.text, borderColor: colors.border }]}
+          placeholder="Search tests..."
+          placeholderTextColor={colors.textSecondary}
+          value={searchQuery}
+          onChangeText={setSearchQuery}
+          autoCapitalize="none"
+          autoCorrect={false}
+          clearButtonMode="while-editing"
+        />
+        {(searchQuery.length > 0 || statusFilter !== 'all') && (
+          <Text style={styles.searchResultsText}>
+            Showing {filteredTests.length} of {tests.length} tests
+          </Text>
+        )}
+      </View>
+
+      <View style={styles.filterContainer}>
+        <View
+          style={[
+            styles.filterSegmentedControl,
+            {
+              backgroundColor: colors.segmentedControlBackground,
+            },
+          ]}
+        >
+          <FilterButton
+            label="All"
+            count={testCounts.total}
+            isActive={statusFilter === 'all'}
+            onPress={() => setStatusFilter('all')}
+            colors={colors}
+          />
+          <FilterButton
+            label="✅ Passed"
+            count={testCounts.passed}
+            isActive={statusFilter === 'passed'}
+            onPress={() => setStatusFilter('passed')}
+            colors={colors}
+          />
+          <FilterButton
+            label="❌ Failed"
+            count={testCounts.failed}
+            isActive={statusFilter === 'failed'}
+            onPress={() => setStatusFilter('failed')}
+            colors={colors}
+          />
+          <FilterButton
+            label="📱 Pending"
+            count={testCounts.pending}
+            isActive={statusFilter === 'pending'}
+            onPress={() => setStatusFilter('pending')}
+            colors={colors}
+          />
+        </View>
+      </View>
```

Comment:
Could this also be moved to a separate file/component and just pass `testCounts` and `statusFilter`? I wanna keep this file here a bit cleaner if possible, you decide